WcaOnRails/spec/requests/api_competitions_spec.rb

```diff
@@ -5,6 +5,46 @@
 RSpec.describe "API Competitions" do
   let(:headers) { { "CONTENT_TYPE" => "application/json" } }
 
+
+  describe "GET #results" do
+    let!(:competition) { FactoryBot.create :competition, :visible }
+
+    it "renders properly" do
+      get api_v0_competition_results_path(competition)
+      expect(response).to be_successful
+      json = JSON.parse(response.body)
+      if json[0]
+        expect(json[0]["competitionId"]).to eq competition
+      end
+    end
+  end
+
+  describe "GET #competitors" do
+    let!(:competition) { FactoryBot.create :competition, :visible }
+
+    it "renders properly" do
+      get api_v0_competition_competitors_path(competition)
+      expect(response).to be_successful
+      json = JSON.parse(response.body)
+      if json[0]
+        expect(json[0]["class"]).to eq "person"
+      end
+    end
+  end
+
+  describe "GET #registrations" do
+    let!(:competition) { FactoryBot.create :competition, :visible }
+
```

Comment:
Here I'd go for two registrations like so:
```rb
let!(:accepted_registration) { FactoryBot.create :registration, :accepted, competition: competition }
let!(:pending_registration) { FactoryBot.create :registration, competition: competition }
```
And then a check that only accepted registrations get listed (e.g. `expect(json.map(&:id)).to eq [accepted_registration.id]`).